IFRCGo/go-frontend
# Fix/1189 validation

## Issue
#1189 

## Description
This adds additional validation to start date and changes numbers to integers

I also fixed an issue I encountered where I couldn't clear the country dropdown because `updateDistricts` was expecting a value from countries

This resolves everything on the ticket with the exception of the date validation. See issue for further conversation.

### Testing Instructions
https://ifrc-go-fix-1189-validation.surge.sh/reports/new

should error on:
- adding an invalid start date
- adding a number in step 2 (or others) with decimal 

should pass on:
- adding a valid start date
- adding a country, clearing the country and adding again
- adding an integer for fields in step 2 (or others)
